ui/src/app/applications/components/application-sync-panel/application-sync-panel.tsx

```diff
@@ -172,9 +172,27 @@ export const ApplicationSyncPanel = ({application, selectedResource, hide}: {app
                                                 .filter(item => !item.hook)
                                                 .map((item, i) => {
                                                     const resKey = nodeKey(item);
+                                                    const contentStart = resKey.substr(0, Math.floor(resKey.length / 2));
+                                                    let contentEnd = resKey.substr(-Math.floor(resKey.length / 2));
+                                                    // Fix Latin letters being rendered left to right on ellipsis with direction set to rtl
+                                                    const indexOfFirstLetter = /[a-z]/i.exec(contentEnd).index;
+                                                    contentEnd = contentEnd.slice(indexOfFirstLetter);
```

Comment:
I'm puzzled by this.
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

Afaict, this regex: `/[-0-9a-zA-Z.]{0,253}/` allows more things than kubernetes does.

Does ArgoCD support translations at all?

Reply:
Thanks for the review. Could you elaborate what translation did you mean? 

Code in line 178 and line 179 is to avoid contentEnd string which starts with special characters such as `/`, `-` and `.` being placed to the rightmost of contentEnd string when rendered with direction property set to rtl value. For example, if contentEnd is `/abcd/argo-cd/test/example`, without the fix it would be rendered as `...abcd/argo-cd/test/example/`.

Reply:
I'm mostly interested in line 177.

`RTL` relates to Arabic and Hebrew rendering (namely that logical elements are rendered Right-to-Left). Typically this only happens if you have a "localization" of an application, typically performed by translating the user interface.

For example:
https://www.google.co.il/?hl=he
![image](https://user-images.githubusercontent.com/2119212/129104735-6f6fcb73-7838-4a3c-bac5-a7b81d16153b.png)

Given that as far as I can tell, it is not possible to view ArgoCD in Arabic or Hebrew, and given that kubernetes resources can't have non ascii (~Latin) characters, a reference to RTL seems out of place.

https://developer.mozilla.org/en-US/docs/Glossary/rtl

Reply:
Simply put, do you have an example of RTL actually breaking? If so, a picture here would be helpful.

Reply:
Sure, I added this only because I observed the weird extra `/`, `-` and `.` at the end of the contentEnd string. Here's an example with out the two lines code. Note that the content-end attribute has the value of `/CustomResourceDefinition/...`, and then the `/` got moved to the end of the contentEnd of the rendered resource string. Did some research on this issue and it turned out `/`, `-` and `.` are considered Latin characters which will be put in the rightmost when use direction of right-to-left. So I made the change to make the contentEnd string starts with the first English letter instead the original string.

<img width="570" alt="Screen Shot 2021-08-12 at 10 06 59 AM" src="https://user-images.githubusercontent.com/26255262/129212337-51174a1e-488b-4066-a347-1ede7ad9fb7b.png">
<img width="406" alt="Screen Shot 2021-08-12 at 10 07 13 AM" src="https://user-images.githubusercontent.com/26255262/129212391-998a82ce-d308-4ab1-8be7-132d15a56ada.png">